src/connector/src/source/iceberg/mod.rs

```diff
@@ -86,25 +110,74 @@ impl SplitMetaData for IcebergSplit {
 }
 
 #[derive(Debug, Clone)]
-pub struct IcebergSplitEnumerator {}
+pub struct IcebergSplitEnumerator {
+    config: IcebergConfig,
+}
 
 #[async_trait]
 impl SplitEnumerator for IcebergSplitEnumerator {
     type Properties = IcebergProperties;
     type Split = IcebergSplit;
 
     async fn new(
-        _properties: Self::Properties,
+        properties: Self::Properties,
         _context: SourceEnumeratorContextRef,
     ) -> ConnectorResult<Self> {
-        Ok(Self {})
+        let iceberg_config = properties.to_iceberg_config();
+        Ok(Self {
+            config: iceberg_config,
+        })
     }
 
     async fn list_splits(&mut self) -> ConnectorResult<Vec<Self::Split>> {
+        // Iceberg source does not support streaming queries
         Ok(vec![])
     }
 }
 
+impl IcebergSplitEnumerator {
+    pub async fn list_splits_batch(
+        &self,
+        batch_parallelism: usize,
+    ) -> ConnectorResult<Vec<IcebergSplit>> {
+        let table = self.config.load_table().await?;
+        let snapshot_id = table.current_table_metadata().current_snapshot_id.unwrap();
```

Comment:
Why not use `snapshot_id` in `IcebergSplitEnumerator `?

Reply:
`IcebergSplitEnumerator` should share the same life cycle as the iceberg source, but each batch read has its own `snapshot_id`.